tensorflow/python/distribute/input_lib_type_spec_test.py

```diff
@@ -450,6 +450,89 @@ def f(v):
       f(v)
     self.assertEqual(self.trace_count, 1)
 
+  @combinations.generate(
+      combinations.combine(
+          mode=["eager"],
+          tf_api_version=2,
+          distribution=[
+              strategy_combinations.mirrored_strategy_with_two_gpus,
+              strategy_combinations.mirrored_strategy_with_cpu_1_and_2,
+              strategy_combinations.mirrored_strategy_with_gpu_and_cpu,
+          ],
+          enable_get_next_as_optional=[True, False]))
+  def testTypeSpecForPerReplicaOptions(self, distribution, enable_get_next_as_optional):
+
+    fname1 = os.path.join(self.get_temp_dir(), "1.txt")
+    _create_text_file(fname1, 5)
+    fname2 = os.path.join(self.get_temp_dir(), "2.txt")
+    _create_text_file(fname2, 9)
+
+    def dataset_fn(input_context):
+      dataset = dataset_ops.DatasetV2.from_tensor_slices([fname1, fname2])
+      dataset = dataset.shard(input_context.num_input_pipelines,
+                              input_context.input_pipeline_id)
+      return readers.TextLineDatasetV2(dataset).map(
+          string_ops.string_to_number).batch(
+              input_context.get_per_replica_batch_size(4))
+    
+    options = distribute_lib.InputOptions(
+        experimental_place_dataset_on_device = True,
```

Comment:
can we add combinations to test for `experimental_place_dataset_on_device=False` and `experimental_prefetch_to_device=True` as well?

Reply:
addressed.